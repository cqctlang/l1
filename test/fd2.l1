orig = stdout;

define r(len){
       fprintf(orig, "reading %d bytes from stdout\n", len);
       return read(orig, len);
}

define w(buf){
       fprintf(orig, "writing %d bytes to stdout\n", strlen(buf));
       return write(orig, buf);
}

define c(){
       fprintf(orig, "closing stdout!\n");
       return close(orig);
}

stdout = mkfd(r,w,c);

printf("hi!\n");
close(stdout);
