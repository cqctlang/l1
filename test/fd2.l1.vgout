writing 4 bytes to stdout
hi!
closing stdout!
