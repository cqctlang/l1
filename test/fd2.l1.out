entry: var params, 1 locals, 0 temps
r: 1 params, 0 locals, 0 temps
w: 1 params, 0 locals, 0 temps
c: 0 params, 0 locals, 0 temps
writing 4 bytes to stdout
hi!
closing stdout!
